### ISSUE TITLE
EKS: need to be able to define worker groups with different instance types

### ISSUE THREAD

**Issue by lanzafame** [protocol/infra-team#28]
Currently, the EKS module only allows you to define a single kind of worker group, which makes it easy to configure the defaults of a worker group, see [here](https://github.com/protocol/infra-team/blob/0c4d86e93ca56e231a93d7194db013a3a2302f2b/bootstrap_terraform/shared_modules/eks/main.tf#L39-L51) But it means we are restricted to just one kind of instance type.

**Comment by eefahy** [comment]
Hrmmm... but wouldn’t that be just another worker group? IIRC, worker groups are tied to an ASG which means they would have a launch config and therefore just one instance type.

**Comment by lanzafame** [comment]
Correct it is another worker group, but the eks takes a list of worker
groups and the way we have defined the variables that we pass through we
can only define one

**Comment by eefahy** [comment]
Oh right, gotcha. I think this wrapper module might be more trouble than it’s worth. You should prob just use the upstream module directly in your repo?

**Comment by lanzafame** [comment]
Yeah that may be the answer. Shame. Well until 0.12 of terraform comes out
and then we can use dynamic nested blocks 😃
